ciao/arch/architecture.md

```diff
@@ -0,0 +1,422 @@
+binary header followed by an optional YAML encoded payload.  SSNTP messages are
+asynchronous.  This means that the sender of a message does not normally wait
+for a response to the message once it has been sent.  The protocol is purposely
+designed this way to mirror the communication architecture of the Cloud
+Integrated Advanced Orchestrator (CIAO).  In a CIAO cluster most of the control
+plane messages are status and statistic messages sent from the compute and
+network nodes by the launcher services to controller.  These messages are sent
+either periodically or when some significant event occurs on the node.  The
+messages are not solicited by controller nor does launcher care whether
+controller correctly receives them.  The asynchronous nature of the majority of
+the control plane communication is best served by an asynchronous protocol, a
+role which SSNTP nicely fulfils.
+
+Note that in some circumstances a response is required to an SSNTP message.  For
+example, consider the case in which controller sends a message to launcher to
+start a new instance.  Controller is actually interested in whether the instance
+started correctly, and if it failed, the reason for its failure.  In this
+case launcher will send controller a start failure message for any instances
+that failed to launch.
+
+For more information about SSNTP the reader is referred to
+[SSNTP](https://github.com/ciao-project/ciao/blob/master/ssntp/README.md).
+
+## Networking
+
+The Cloud Integrated Advanced Orchestrator is supplied with an innovative
+Software Defined Network (SDN) solution that provides simple to use and secure
+networking.  Highlights of this SDN are as follows:
+
+- Each tenant gets its own private network.
+- VM and container instances communicate seamlessly over this private network.
+- Instances can discover each other by performing DNS lookups against tenant local DNS servers.
+- Instances can communicate with the outside world, but by default, cannot be accessed
+  by entities outside the cluster or running in different tenants.
+
+All this functionality comes for free and requires no configuration by the end user.
+
+Instances can of course be made accessible to the outside world and to other
+tenants but setting this up does require some input by the end user.  In short
+the user needs to create a pool of external IP addresses, essentially a range of
+IP addresses that have been allocated for use by the cluster, and then assign
+these IP addresses to the instances that need to be exposed.  This can be done
+by issuing a few simple commands as illustrated in the
+[Creating your first cloud app tutorial](https://ciao-project.github.io/tutorial.html#creating-an-external-ip).
+
+### The Compute Node Concentrator Instance (CNCI)
+
+At the heart of the Cloud Integrated Advanced Orchestrator's networking model is
+the CNCI.  The CNCI is responsible for ensuring that instances can communicate
+both with each other and with the outside world.  In short it
```

Comment:
Your grammar style for the list below is different to the one above. I personally prefer:

0. An introduction with a colon,
1. Each line to begin with a capital letter,
2. And end with a comma,
3. Except for the last that should end with a period.